Compilation for ARM M1 Macbook fails

Hi there!

Compilation via

```shell
pip install -e . 
```

fails on Mac in a few places.

first:

```shell
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/lucanaef/mambaforge/include -arch arm64 -fPIC -O2 -isystem /Users/lucanaef/mambaforge/include -arch arm64 -Imdtraj/rmsd -Imdtraj/rmsd/include -I/Users/lucanaef/mambaforge/lib/python3.9/site-packages/numpy/core/include -I/Users/lucanaef/mambaforge/include/python3.9 -c mdtraj/rmsd/src/theobald_rmsd.cpp -o build/temp.macosx-11.0-arm64-3.9/mdtraj/rmsd/src/theobald_rmsd.o -O3 -funroll-loops -Wno-unused-function -Wno-unreachable-code -Wno-sign-compare
In file included from mdtraj/rmsd/src/theobald_rmsd.cpp:44:
mdtraj/rmsd/src/theobald_rmsd_arm.h:89:29: error: expected '(' for function-style cast or type construction
```

This can be fixed by injecting `-std=c++11` into the `setup.py` (not sure where the proper place to put this is but at least gets rid of it for now.

Next fails:

```shell
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/lucanaef/mambaforge/include -arch arm64 -fPIC -O2 -isystem /Users/lucanaef/mambaforge/include -arch arm64 -Imdtraj/geometry/include -Imdtraj/geometry/src/kernels -I/Users/lucanaef/mambaforge/lib/python3.9/site-packages/numpy/core/include -I/Users/lucanaef/mambaforge/include/python3.9 -c mdtraj/geometry/src/dssp.cpp -o build/temp.macosx-11.0-arm64-3.9/mdtraj/geometry/src/dssp.o -O3 -funroll-loops -Wno-unused-function -Wno-unreachable-code -Wno-sign-compare
    In file included from mdtraj/geometry/src/dssp.cpp:19:
    In file included from mdtraj/geometry/include/vectorize.h:36:
    mdtraj/geometry/include/vectorize_neon.h:38:10: fatal error: 'sys/auxv.h' file not found
    #include <sys/auxv.h>
             ^~~~~~~~~~~~
    1 error generated.
```

It looks like this lib isn't available on ARM OS X

Using:

python 3.9
apple clang version 12.0.5 (clang-1205.0.22.11)
Target: arm64-apple-darwin20.6.0
2020 MacBook Air running Big Sur

--- a/basesetup.py
+++ b/basesetup.py
@@ -50,7 +50,7 @@ class CompilerDetection(object):
             self.openmp_enabled, openmp_needs_gomp = self._detect_openmp()
         self.sse3_enabled = self._detect_sse3() if not self.msvc else True
         self.sse41_enabled = self._detect_sse41() if not self.msvc else True
-        self.neon_enabled = self._detect_neon() if not self.msvc else False        
+        self.neon_enabled = self._detect_neon() if not self.msvc else False
 
         self.compiler_args_sse2 = ['-msse2'] if not self.msvc else ['/arch:SSE2']
         self.compiler_args_sse3 = ['-mssse3'] if (self.sse3_enabled and not self.msvc) else []
@@ -83,7 +83,7 @@ class CompilerDetection(object):
         if self.msvc:
             self.compiler_args_opt = ['/O2']
         else:
-            self.compiler_args_opt = ['-O3', '-funroll-loops']
+            self.compiler_args_opt = ['-O3', '-funroll-loops', '--std=c++11']
         print()
         self._is_initialized = True
 

--- a/mdtraj/geometry/include/vectorize_neon.h
+++ b/mdtraj/geometry/include/vectorize_neon.h
@@ -34,7 +34,7 @@
 
 #ifdef __ANDROID__
 #include <cpu-features.h>
-#else
+#elif !defined(__APPLE__)
 #include <sys/auxv.h>
 #include <asm/hwcap.h>
 #endif
@@ -53,7 +53,9 @@ float32x4_t log_ps(float32x4_t);
  * Determine whether ivec4 and fvec4 are supported on this processor.
  */
 static bool isVec4Supported() {
-#ifdef __ANDROID__
+#ifdef __APPLE__
+    return true;
+#elif defined(__ANDROID__)
     uint64_t features = android_getCpuFeatures();
     return (features & ANDROID_CPU_ARM_FEATURE_NEON) != 0;
 #elif defined(__ARM__)

